src/main/java/net/snowflake/ingest/streaming/internal/ParquetFlusher.java

```diff
@@ -160,29 +169,50 @@ private SerializationResult serializeFromJavaObjects(
   }
 
   private void addFileIdToMetadata(
-      String filePath, long chunkStartOffset, Map<String, String> metadata) {
+      String filePath,
+      long chunkStartOffset,
+      Map<String, String> metadata,
+      Optional<String> customFileIdForTesting) {
     // We insert the filename in the file itself as metadata so that streams can work on replicated
     // mixed tables. For a more detailed discussion on the topic see SNOW-561447 and
     // http://go/streams-on-replicated-mixed-tables,  and
     // http://go/managed-iceberg-replication-change-tracking
     // Using chunk offset as suffix ensures that for interleaved tables, the file
     // id key is unique for each chunk. Each chunk is logically a separate Parquet file that happens
     // to be bundled together.
-    if (chunkStartOffset == 0) {
-      metadata.put(
-          enableIcebergStreaming
-              ? Constants.ASSIGNED_FULL_FILE_NAME_KEY
-              : Constants.PRIMARY_FILE_ID_KEY,
-          StreamingIngestUtils.getShortname(filePath));
-    } else {
+    if (enableIcebergStreaming) {
       Preconditions.checkState(
-          !enableIcebergStreaming, "Iceberg streaming is not supported with non-zero offsets");
-      String shortName = StreamingIngestUtils.getShortname(filePath);
-      final String[] parts = shortName.split("\\.");
-      Preconditions.checkState(parts.length == 2, "Invalid file name format");
-      metadata.put(
-          Constants.PRIMARY_FILE_ID_KEY,
-          String.format("%s_%d.%s", parts[0], chunkStartOffset, parts[1]));
+          chunkStartOffset == 0, "Iceberg streaming is not supported with non-zero offsets");
+    }
+    String key =
+        enableIcebergStreaming
+            ? Constants.ASSIGNED_FULL_FILE_NAME_KEY
+            : Constants.PRIMARY_FILE_ID_KEY;
+    String value =
+        customFileIdForTesting.orElseGet(
+            () -> {
+              String shortName = StreamingIngestUtils.getShortname(filePath);
+              if (chunkStartOffset == 0) {
+                return shortName;
+              } else {
+                final String[] parts = shortName.split("\\.");
+                Preconditions.checkState(parts.length == 2, "Invalid file name format");
+                return String.format("%s_%d.%s", parts[0], chunkStartOffset, parts[1]);
+              }
+            });
+    metadata.put(key, value);
+  }
+
+  private void overrideSdkVersionInMetadata(
```

Comment:
would it make sense to keep `addFileIdToMetadata` untouched to reduce risk
and do `customFileIdForTesting` in `overrideSdkVersionInMetadata`?
```
customFileIdForTesting.ifPresent(fid ->
  metadata.put(
          enableIcebergStreaming
              ? Constants.ASSIGNED_FULL_FILE_NAME_KEY
              : Constants.PRIMARY_FILE_ID_KEY,
          fid);
);
```